source/agora/common/crypto/Schnorr.d

```diff
@@ -230,31 +241,59 @@ public Signature sign (T) (
 public bool verify (T) (const ref Point X, const ref Signature sig, auto ref T data)
     nothrow @nogc @trusted
 {
-    Sig s = Sig.fromBlob(sig);
-    // Compute the challenge and reduce the hash to a scalar
-    Scalar c = hashFull(Message!T(X, s.R, data));
-    // Compute `R + c*X`
-    Point RcX = s.R + (c * X);
-    /// Compute `s.G`
-    auto S = s.s.toPoint();
-    return S == RcX;
+    try {
+        Sig s = Sig.fromBlob(sig);
+        if (s.s.isValid()) {
+            /// Compute `s.G`
+            auto S = s.s.toPoint();
+            // Compute the challenge and reduce the hash to a scalar
+            Scalar c = hashFull(Message!T(X, s.R, data));
+            // Compute `R + c*X`
+            Point RcX = s.R + (c * X);
+            return S == RcX;
+        } else {
+            return false;
+        }
+    } 
+    catch(Exception e) 
+    {
+        debug { import std.stdio : writeln; try { writeln("verify signature exception: %s", e.message); } catch (Exception) {} }
+        return false;
+    }
 }
 
 ///
 nothrow @nogc @safe unittest
 {
     Scalar key = Scalar(`0x074360d5eab8e888df07d862c4fc845ebd10b6a6c530919d66221219bba50216`);
     Pair kp = Pair(key, key.toPoint());
-    auto signature = sign(kp, "Hello world");
-    assert(verify(kp.V, signature, "Hello world"));
+    static immutable string message = "Valid signing test with valid scalar";
+    auto signature = sign(kp, message);
+    assert(verify(kp.V, signature, message));
 }
 
 nothrow @nogc @safe unittest
 {
-    Scalar key = Scalar(`0x074360d5eab8e888df07d862c4fc845ebd10b6a6c530919d66221219bba50216`);
-    Pair kp = Pair(key, key.toPoint());
-    auto signature = sign(kp, "Hello world.");
-    assert(!verify(kp.V, signature, "Hello world"));
+    Pair kp = Pair.fromScalar(Scalar(`0x0000000000000000000000000000000000000000000000000000000000000001`));
+    static immutable string message = "Valid with scalar value 1";
```

Comment:
Should move it to the unittest comment. Since this one is not really important for the documentation we can use `//` instead of `///` for the comment - this means there will be no documentation generated for the test (but will still be visible in the editor of course).

```
// Valid with scalar value 1
nothrow @nogc @safe unittest
{
    ...
}
```